configure.ac

```diff
@@ -612,7 +612,7 @@ AC_CANONICAL_TARGET
 # Find a C compiler.
 # We also need to do this before the F77 and FC test to ensure that we
 # find the C preprocessor reliably.
-PAC_PROG_CC
+AC_PROG_CC
```

Comment:
What is the default compiler name that this macro searches? Maybe we should keep the compiler name list used in `PAC_PROG_CC`. (`icc pgcc xlc xlC pathcc gcc clang cc]`).

Reply:
It picks GNU by default.  Is there a reason to force that order?  This makes it somewhat different from other systems making composability harder.

Reply:
What I understand is we were searching from a larger set of compilers (intel, IBM, PGI) than just GNU, so the macro needed the list. Not sure why those compilers are selected in the first place as the code predates me.

I guess what this change say is if the users want a different compiler, they should always set `CC` environment variable than relying on the `configure` to detect for them.

Reply:
Yes, users cannot rely on that list anyway.  Plus the non-default order means that MPICH might pick one compiler and hwloc might pick a different compiler, creating a mess of the configuration.